### PR TITLE
refactor: rename rmrk2 to ksm

### LINKED ISSUE #5418: Rename rmrk2 prefix to ksm in static
Continue from 
- #5396

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #5418

#### Before submitting pull request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [ ] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=<My_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address>)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by vikiival** [comment]
The thing is that once marck will be 100% ready we plan to:
merge rubick@008 & marck@001 into rubick@9. 
This will use only one prefix. 

I am aware that renaming RMRK to KSM should be an issue. Because it is hardly baked into code. 

Therefore proposal to rename RMRK2 to KSM. Should not be an issue. 
Rmrk2 is currently in alpha/internal testing so breaking urls should not be an issue 🤗

**Comment by exezbcz** [comment]
and what would be shown in the gallery detail? Also kusama? 
image


- If I understand that correctly rmrk1 is kusama as well - so we won't distinguish between RMRK1 & RMRK2

**Comment by vikiival** [comment]
> and what would be shown in the gallery detail? Also kusama? 
> 
> image
> 
> 
> 
> 
> 
> - If I understand that correctly rmrk1 is kusama as well - so we won't distinguish between RMRK1 & RMRK2
> 
> 

Yup both are kusama based NFTs, they use same protocol, but different version (obtained from indexer) and as I mentioned they will be merged together soon.

**Comment by exezbcz** [comment]
@vikiival oki makes sense, would be nice if its merged together, since both version dont have offers anyway :D 🚀 

thanks

**Comment by roiLeo** [comment]
> Yup both are kusama based NFTs, they use same protocol, but different version (obtained from indexer) and as I mentioned they will be merged together soon.

Statemine be like
crycat-crying-cat

**Comment by yangwao** [comment]
> merge rubick@008 & marck@001 into rubick@9.
> This will use only one prefix.

how we will distinguish nfts on frontend if it's rmrk1 and rmrk2? @vikiival

**Comment by vikiival** [comment]
> how we will distinguish nfts on frontend if it's rmrk1 and rmrk2?

We have version stored on the Subsquid

```json
{
  "data": {
    "nftEntities": [
      {
        "name": "Mohawk",
        "version": "2.0.0"
      },
      {
        "name": "Earring",
        "version": "2.0.0"
      },
      {
        "name": "MOONSAMA SPECIAL",
        "version": "2.0.0"
      },
      {
        "name": "MOONSAMA SPECIAL",
        "version": "2.0.0"
      },
      {
        "name": "Referendum 201",
        "version": "2.0.0"
      }
    ]
  }
}
```

version 1 would be `1.0.0`

File: libs/static/src/chains.ts [hunk 1]
```diff
@@ -44,29 +44,32 @@ export const CHAINS: Config<ChainProperties> = {
...
   }))
 }
 
-export const oldKsm = (prefix: BackwardPrefix): Prefix =>
-  prefix === 'rmrk' ? 'ksm' : prefix
+// TODO: our url prefix still rmrk2
+// e.g: https://kodadot.xyz/rmrk2/explore/items
+export const rmrk2ToKsm = (prefix: BackwardPrefix): Prefix => {
+  return prefix === 'rmrk2' ? 'ksm' : prefix
+}
```
Comment (vikiival) [inline libs/static/src/chains.ts:75]: I would deprecate this there is no many urls in the wild

Comment (vikiival) [reply libs/static/src/chains.ts:75]: > this section is to help transform CHAINS['rmrk2'] to CHAINS['ksm'].

Yup I understand. What I do not get is why do we need that :D

**Comment by yangwao** [comment]
anything we are waiting for?

**Comment by vikiival** [comment]
> anything we are waiting for? 
Yup I would like to deprecate `/rmrk2` urls so anything which will run on KSM will have /ksm prefix in url

**Comment by yangwao** [comment]
yup, until we are showing in description something like ksm/rmrk1 and ksm/rmrk2 it's good to go

File: libs/static/src/chains.ts (hunk 1 above)
Comment (vikiival) [reply libs/static/src/chains.ts:75]: so let's remove this a per comment from @yangwao

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
I'll make follow-up issue
